### PR TITLE
[StyleFix][Update] Documentation

### PR DESCRIPTION
**Pull request by mzanrosso**
Changes:
1. Removed duplicate documentation in lib.rs 
2. Updated readme file with latest improvements and changes

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,57 +1,3 @@
-//! # reedline `\|/`
-//! A readline replacement written in Rust
-//!
-//! ## Example (Simple REPL)
-//!
-//! ```rust,no_run
```
Comment (sholderbach) [inline src/lib.rs:6]: `lib.rs` doctests should at least compile every time.

File: src/lib.rs [hunk 2]
```diff
@@ -1,57 +1,3 @@
...
-//! [PR](https://github.com/jonathandturner/reedline/pulls)!
-//!
-//! * `[x]` Basic unicode grapheme aware cursor editing.
-//! * `[x]` Configurable prompt
-//! * `[x]` Basic EMACS-style editing shortcuts.
-//! * `[ ]` Advanced multiline unicode aware editing.
-//! * `[x]` Configurable keybindings.
-//! * `[x]` Basic system integration with clipboard or optional stored history file.
-//! * `[ ]` Content aware highlighting or validation.
-//! * `[ ]` Autocompletion.
```
Comment (sholderbach) [inline src/lib.rs:46]: I think with the recent revisions `rustdoc` is now able to parse the GH flavored checkbox lists.

**Review by sholderbach — COMMENTED** [review]
As the doc-comments of `lib.rs` are used to create the main page of the `rustdoc` documentation for [docs.rs](http://docs.rs) we should either keep it up to date or include it from `README.md`

File: src/lib.rs (hunk 1 above)
Comment (sholderbach) [reply src/lib.rs:6]: `cargo test` passed for me with the latest changes.

File: src/lib.rs [hunk 3]
```diff
@@ -1,57 +1,174 @@
...
+//!
+//! ```rust,no_run
+//! // Create a reedline object with custom edit mode
+//!
+//! use reedline::{EditMode, Reedline};
+//!
+//! let mut line_editor = Reedline::new().with_edit_mode(
+//!   EditMode::ViNormal, // or EditMode::Emacs
+//! );
+//! ```
```
Comment (sholderbach) [inline src/lib.rs:66]: As vi support is currently rudimentary we might want to push this example a bit down. (Also most shells boot into `ViInsert`)

**Comment by mzanrosso** [comment]
@sholderbach - Let me know if you see something that i can improve in the readme file

File: src/lib.rs [hunk 4]
```diff
@@ -1,57 +1,177 @@
...
+//!             Signal::CtrlL => {
+//!                 line_editor.clear_screen().unwrap();
+//!             }
 //!         }
 //!     }
 //! }
+//! ```
+//! ## Integrate with custom Keybindings
+//!
+//! ```rust,no_run
```
Comment (sholderbach) [inline src/lib.rs:34]: Some tests might actually not hit the terminal and don't run in an endless I/O loop like the core example. So we might carefully consider at some point switching to live tests checking at least the constructors.

**Review by sholderbach — COMMENTED** [review]
Thanks for the quick fix! When running `cargo doc` there are a few warnings due to changed references but we can fix them separately

File: src/lib.rs (hunk 4 above)
Comment (mzanrosso) [reply src/lib.rs:34]: Okay

File: src/lib.rs (hunk 3 above)
Comment (mzanrosso) [reply src/lib.rs:66]: Done.

**Comment by sholderbach** [comment]
Looks great to me! Let me know if you are ready to merge (or if @jntrnr wants to chime in)

**Comment by mzanrosso** [comment]
@sholderbach - Yeah for me it's okay to merge, maybe some comment could be more explicit but it's fine even like this.

**Pull request merged**